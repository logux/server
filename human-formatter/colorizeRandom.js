const COLORS = ['red', 'green', 'yellow', 'blue', 'magenta', 'cyan']

function randomToColorName(str) {
  let res = 0
  for (let i = 0; i < str.length; i += 1) {
    let code = str.charCodeAt(i)
    res += code
  }
  return COLORS[res % COLORS.length]
}

export function colorizeRandom(c, str) {
  let color = randomToColorName(str)
  return c[color](str)
}
