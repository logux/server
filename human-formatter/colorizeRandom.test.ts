import pico from 'picocolors'

import { colorizeRandom } from './colorizeRandom.js'

let c = pico.createColors(true)

it('should colorize it red', () => {
  let result = colorizeRandom(c, '10:client:6KBKxMBs')
  expect(result).toMatchSnapshot()
})

it('should colorize in green', () => {
  let result = colorizeRandom(c, '10:client:GX0kXIgq')
  expect(result).toMatchSnapshot()
})

it('should colorize in yellow', () => {
  let result = colorizeRandom(c, '10:client:V40axO_O')
  expect(result).toMatchSnapshot()
})

it('should colorize blue', () => {
  let result = colorizeRandom(c, '10:client:Eizt7-PG')
  expect(result).toMatchSnapshot()
})

it('should colorize in magenta', () => {
  let result = colorizeRandom(c, '10:client:44FWEisN')
  expect(result).toMatchSnapshot()
})

it('should colorize cyan', () => {
  let result = colorizeRandom(c, '10:client:uFPMXJDO')
  expect(result).toMatchSnapshot()
})
