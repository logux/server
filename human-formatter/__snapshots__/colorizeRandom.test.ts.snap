// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should colorize blue 1`] = `"[34m10:client:Eizt7-PG[39m"`;

exports[`should colorize cyan 1`] = `"[36m10:client:uFPMXJDO[39m"`;

exports[`should colorize in green 1`] = `"[32m10:client:GX0kXIgq[39m"`;

exports[`should colorize in magenta 1`] = `"[35m10:client:44FWEisN[39m"`;

exports[`should colorize in yellow 1`] = `"[33m10:client:V40axO_O[39m"`;

exports[`should colorize it red 1`] = `"[31m10:client:6KBKxMBs[39m"`;
